app/controllers/api/metrics_controller.rb

```diff
@@ -23,27 +23,43 @@ def query_series
                .between(from: from, to: to)
                .by_base_path(format_base_path_param)
                .run
-    if Metric.is_edition_metric?(metric)
-      series
-        .with_edition_metrics
-        .order('dimensions_dates.date asc')
-        .pluck(:date, "facts_editions.#{metric}").to_h
-    else
-      series
-        .order('dimensions_dates.date asc')
-        .pluck(:date, metric).to_h
+
+    metric_names = format_metric_names
+
+    series_array = series.with_edition_metrics
+                    .order('dimensions_dates.date asc')
+                    .pluck(:date, *metric_names)
+
+    dates_to_metrics(series_array)
+  end
+
+  def format_metric_names
+    metrics = params[:metrics] ? params[:metrics] : [params[:metric]]
+    metrics.map! do |metric|
+      Metric.is_edition_metric?(metric) ? "facts_editions.#{metric}" : metric
     end
   end
 
+  def dates_to_metrics(series_array)
```

Comment:
Doing data transformation in the controller can cause issues in the mid - long term. Could you please move this logic in the the Json view while we set up Serializers?